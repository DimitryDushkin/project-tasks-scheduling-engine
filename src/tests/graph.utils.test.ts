import { makeGraphFromTasks, Graph, makeReverseGraph } from "../graph.utils";
import { tasks } from "./mocks/tasks.mocks";

describe("graph.utils", () => {
  it("should make correct graph from tasks", () => {
    expect(makeGraphFromTasks(tasks)).toMatchSnapshot();
  });

  it("should make correct reverse graph", () => {
    const graph: Graph = new Map([
      ["0", new Set(["1", "3"])],
      ["1", new Set("2")],
      ["2", new Set("3")]
    ]);

    expect(makeReverseGraph(graph)).toMatchSnapshot();
  });
});
