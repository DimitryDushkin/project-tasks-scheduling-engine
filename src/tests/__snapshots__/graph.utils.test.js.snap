// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graph.utils should make correct graph from tasks 1`] = `
Map {
  "0" => Set {
    "1",
    "3",
  },
  "1" => Set {
    "2",
  },
  "2" => Set {
    "3",
  },
  "3" => Set {},
}
`;

exports[`graph.utils should make correct reverse graph 1`] = `
Map {
  "0" => Set {},
  "1" => Set {
    "0",
  },
  "3" => Set {
    "0",
    "2",
  },
  "2" => Set {
    "1",
  },
}
`;
